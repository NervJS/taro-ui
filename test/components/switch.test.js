import Nerv, { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'
import AtSwitch from '../../.temp/components/switch/index'

describe('AtRate Snap', () => {
  it('render initial AtSwitch', () => {
    const componet = renderToString(<AtSwitch />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSwitch -- props title', () => {
    const componet = renderToString(<AtSwitch title='开启中' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSwitch -- props checked', () => {
    const componet = renderToString(<AtSwitch title='开启中' checked />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSwitch -- props border', () => {
    const componet = renderToString(<AtSwitch title='开启中' border={false} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSwitch -- props disabled', () => {
    const componet1 = renderToString(
      <AtSwitch title='开启中' checked disabled />
    )
    expect(componet1).toMatchSnapshot()
    const componet2 = renderToString(<AtSwitch title='开启中' disabled />)
    expect(componet2).toMatchSnapshot()
  })
})

describe('AtSwitch Event', () => {
  it('AtSwitch onChange', () => {
    const onItemClick = jest.fn()
    const component = renderIntoDocument(
      <AtSwitch title='开启中' checked onChange={onItemClick} />
    )
    const items = findDOMNode(component, 'at-switch').querySelectorAll(
      '.at-switch__switch'
    )
    const item0 = items[0]
    Simulate.click(item0)
    expect(onItemClick).toBeCalled()
    expect(onItemClick.mock.calls[0][0].value).toBeFalsy()
  })
})
