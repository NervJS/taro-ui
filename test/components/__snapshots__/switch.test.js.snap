// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtRate Snap render AtSwitch -- props border 1`] = `"<div class=\\"at-switch at-switch--without-border\\"><div class=\\"at-switch__title\\">开启中</div><div class=\\"at-switch__container\\"><div class=\\"at-switch__mask\\"></div><input color=\\"#6190e8\\" class=\\"weui-switch at-switch__switch\\" type=\\"checkbox\\"/></div></div>"`;

exports[`AtRate Snap render AtSwitch -- props checked 1`] = `"<div class=\\"at-switch\\"><div class=\\"at-switch__title\\">开启中</div><div class=\\"at-switch__container\\"><div class=\\"at-switch__mask\\"></div><input color=\\"#6190e8\\" class=\\"weui-switch at-switch__switch\\" checked type=\\"checkbox\\" style=\\"border-color:#6190e8;background-color:#6190e8;\\"/></div></div>"`;

exports[`AtRate Snap render AtSwitch -- props disabled 1`] = `"<div class=\\"at-switch\\"><div class=\\"at-switch__title\\">开启中</div><div class=\\"at-switch__container at-switch--disabled\\"><div class=\\"at-switch__mask\\"></div><input color=\\"#6190e8\\" class=\\"weui-switch at-switch__switch\\" checked type=\\"checkbox\\" style=\\"border-color:#6190e8;background-color:#6190e8;\\"/></div></div>"`;

exports[`AtRate Snap render AtSwitch -- props disabled 2`] = `"<div class=\\"at-switch\\"><div class=\\"at-switch__title\\">开启中</div><div class=\\"at-switch__container at-switch--disabled\\"><div class=\\"at-switch__mask\\"></div><input color=\\"#6190e8\\" class=\\"weui-switch at-switch__switch\\" type=\\"checkbox\\"/></div></div>"`;

exports[`AtRate Snap render AtSwitch -- props title 1`] = `"<div class=\\"at-switch\\"><div class=\\"at-switch__title\\">开启中</div><div class=\\"at-switch__container\\"><div class=\\"at-switch__mask\\"></div><input color=\\"#6190e8\\" class=\\"weui-switch at-switch__switch\\" type=\\"checkbox\\"/></div></div>"`;

exports[`AtRate Snap render initial AtSwitch 1`] = `"<div class=\\"at-switch\\"><div class=\\"at-switch__title\\"></div><div class=\\"at-switch__container\\"><div class=\\"at-switch__mask\\"></div><input color=\\"#6190e8\\" class=\\"weui-switch at-switch__switch\\" type=\\"checkbox\\"/></div></div>"`;
